src/codegen/pass2_generate.cc

```diff
@@ -755,17 +776,14 @@ static CodeList* gen_goswif(
 }
 
 template<typename label_t>
-static const char* gen_gocp_entry(Scratchbuf& buf, bool relative, const std::string& prefix,
+static const char* gen_gocp_entry(Output& output, const std::string& prefix,
         const label_t base_label, const label_t label) {
-    if (relative) {
-        // TODO: port to syntax file and replace hardcoded `int` with `code:type_yytarget`
-        buf.cstr("(int)(").cstr("(char*)").cstr("&&").str(prefix)
-            .label(label).cstr(" - (char*)").cstr("&&")
-            .str(prefix).label(base_label).cstr(")");
-    } else {
-        buf.cstr("&&").str(prefix).label(label);
-    }
-    return buf.flush();
+    const opt_t* opts = output.block().opts;
+    Scratchbuf& buf = output.scratchbuf;
+    const std::string base = buf.str(prefix).label(base_label).flush();
+    const std::string lbl = buf.str(prefix).label(label).flush();
```

Comment:
Instead of creating these temporary strings with unnecessarily long lifetime, can you save options and label references in `GenTargetElem` and output them directly?